python/tests/base.py

```diff
@@ -0,0 +1,115 @@
+import re
```

Comment:
Can we rename this to base_tests.py or something? Also, I see this line in our automated tests in test.yml . Is the coverage report skipping the base.py tests right now? 

        coverage run -m unittest discover -s tests -p '*_tests.py'

Reply:
`base.py` doesn't define any tests, it is just a container for the base classes (ie. `SparkTest`). It does not need to be run by the test runner on its own - it is imported into the other test cases.

Reply:
Got it, looks good.